workers/go/kitchensink/kitchen_sink.go

```diff
@@ -88,10 +101,31 @@ func KitchenSinkWorkflow(ctx workflow.Context, params *kitchensink.WorkflowInput
 			if actionSet == nil {
 				actionSet = sigActions.GetDoActions()
 			}
+
+			receivedID := sigActions.GetSignalId()
+
+			// Handle signal deduplication if signal ID is provided
+			if receivedID != 0 {
+				err := state.handleSignalDeduplication(receivedID)
+				if err != nil {
+					workflow.GetLogger(ctx).Error("error handling signal deduplication", "error", err)
+					retOrErrChan.Send(ctx, ReturnOrErr{nil, err})
+					return
+				}
+			}
+
+			// Execute action set in goroutine for consistent handling
 			workflow.Go(ctx, func(ctx workflow.Context) {
 				ret, err := state.handleActionSet(ctx, actionSet)
 				if ret != nil || err != nil {
 					retOrErrChan.Send(ctx, ReturnOrErr{ret, err})
+					return
+				}
+
+				// Check if all expected signals have been received (only for signals with IDs)
+				if receivedID != 0 && state.validateSignalCompletion() {
+					state.workflowState.Kvs["signals_complete"] = "true"
```

Comment:
Do you even need this in the workflow state kvs?
The workflow should just fail if it did not receive all of the expected signals.